### PR TITLE
Link query real

### PR DESCRIPTION
**Pull request by nevogithub**
Pull request for category links

### CODE REVIEW

File: .vscode/settings.json
Comment (YechiamW3) [inline .vscode/settings.json:1]: should be .gitignored

File: cv_next/app/feed/components/feed.tsx [hunk 1]
```diff
@@ -10,14 +10,43 @@ import Definitions from "@/lib/definitions";
...
 import { FilterPanel, filterValues } from "@/app/feed/components/filterPanel";
 import ReactLoading from "react-loading";
+import { useRouter, useSearchParams, usePathname } from "next/navigation";
+import Categories from "@/types/models/categories";
+
+function toCategory(name: string):number {
+  return Categories.category[name as keyof typeof Categories.category];
+}
+
+function categoryString(num:number){
```
Comment (YechiamW3) [inline cv_next/app/feed/components/feed.tsx:20]: This should sit in Definitions.

File: cv_next/app/feed/components/feed.tsx [hunk 2]
```diff
@@ -60,6 +88,14 @@ export default function Feed() {
   const fetchCvsCallback = useCallback(async () => {
     if (loadMore) {
       const nextPage = page.current + 1;
+      if(optionalCategory){
+        if(filters.categoryId){
+          filters.categoryId  = [...new Set(filters.categoryId.concat(optionalCategory))];
```
Comment (YechiamW3) [inline cv_next/app/feed/components/feed.tsx:93]: Does this reset the categories if they were changed in the search param?
i.e. if I do `/filter?category=abc&category=def` (not sure if that's exactly the syntax here), and then it gets changed to `/filter?category=ghi` will I get set of ["abc","def","ghi"] or just ["ghi"] on the second query?

File: cv_next/app/feed/components/feed.tsx [hunk 3]
```diff
@@ -70,7 +106,17 @@ export default function Feed() {
       } else {
         setLoadMore(false);
       }
+      if(filters.categoryId){
+        params.delete("category");
+        console.log("params before", params);
```
Comment (YechiamW3) [inline cv_next/app/feed/components/feed.tsx:111]: Delete the log. If you want to keep it, put a debug log here.

File: cv_next/app/feed/components/filterPanel.tsx [hunk 4]
```diff
@@ -20,12 +20,12 @@ export const FilterPanel = ({
   onChange: (filters: filterValues) => void;
 }) => {
   const [searchValue, setSearchValue] = useState(defaultFilters.searchValue);
-  const [categoryId, setCategoryId] = useState(defaultFilters.categoryId);
+  const [categoryId, setCategoryId] = useState(defaultFilters.categoryId)
```
Comment (YechiamW3) [inline cv_next/app/feed/components/filterPanel.tsx:23]: Bad prettier config, run again on all the PR files with the proper configuration.

File: cv_next/app/feed/page.tsx [hunk 5]
```diff
@@ -4,10 +4,10 @@ import { handleCurrentUser } from "../actions/user/fetchUserInfo";
...
 
 export default async function Page() {
-  const redirectOutput = await handleCurrentUser("/feed");
-  if (redirectOutput != "/feed") {
-    redirect(redirectOutput);
-  }
+  //const redirectOutput = await handleCurrentUser("/feed");
+  //if (redirectOutput != "/feed") {
+  //  redirect(redirectOutput);
+  //}
```
Comment (YechiamW3) [inline cv_next/app/feed/page.tsx:10]: Don't leave dangling comments.

File: cv_next/server/api/cvs.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
-import "server-only";
+"use server";
+//import "server-only";
```
Comment (YechiamW3) [inline cv_next/server/api/cvs.ts:2]: Again, don't leave dangling comments.

@Aradon267 do we have a git hook for this? Preferably pre-commit

File: cv_next/server/api/cvs.ts [hunk 7]
```diff
@@ -17,8 +20,9 @@ export const revalidate = Definitions.CVS_REVALIDATE_TIME_IN_SECONDS;
  * @return {Promise<CvModel | null>} The retrieved CV or null if not found
  */
 export async function getCvById(cvId: string): Promise<CvModel | null> {
+  'use server';
```
Comment (YechiamW3) [inline cv_next/server/api/cvs.ts:23]: This is not how `use server` works. If you put `use server` on the entire file, the entire file would be referred as server file.

File: cv_next/server/api/cvs.ts [hunk 8]
```diff
@@ -148,21 +155,36 @@ export async function getPaginatedCvs(
...
       .range(from, to - 1);
-
     logger.debug(filters, "filters");
     if (filters) {
       if (filters.searchValue) {
+        let userIdResult = await getUserIdByName(filters.searchValue);
+        filters.searchValue = filters.searchValue.replace(" ", "+");
+        if((userIdResult).ok) {
+          const userIdList = userIdResult.val;
+          console.log(userIdList);
```
Comment (YechiamW3) [inline cv_next/server/api/cvs.ts:179]: console.log..

File: cv_next/server/api/supabaseHelper.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import "server-only";
+//import "server-only";
```
Comment (YechiamW3) [inline cv_next/server/api/supabaseHelper.ts:1]: Why?

File: cv_next/server/api/cvs.ts (hunk 6 above)
Comment (YechiamW3) [reply cv_next/server/api/cvs.ts:2]: Also, why remove the import server only?

File: cv_next/server/api/users.ts [hunk 10]
```diff
@@ -1,168 +1,31 @@
...
+
+import SupabaseHelper from "./supabaseHelper";
+import { Ok, Err } from "@/lib/utils";
+import { Tables, ProfileKeys } from "@/lib/supabase-definitions";
+
+export async function getUserIdByName(
+  fullName: string
+): Promise<Result<string[], string>> {
+  try {
+    console.log(fullName);
```
Comment (YechiamW3) [inline cv_next/server/api/users.ts:11]: Make sure to catch them all

File: cv_next/server/api/cvs.ts (hunk 6 above)
Comment (Aradon267) [reply cv_next/server/api/cvs.ts:2]: I remember that I added pre-commit, and also an eslint rule to warn at console.log, will look into it

File: cv_next/app/feed/components/feed.tsx (hunk 2 above)
Comment (nevogithub) [reply cv_next/app/feed/components/feed.tsx:93]: I think it resets the categories fully

File: cv_next/app/feed/components/feed.tsx (hunk 3 above)
Comment (nevogithub) [reply cv_next/app/feed/components/feed.tsx:111]: ok

File: cv_next/server/api/cvs.ts (hunk 7 above)
Comment (nevogithub) [reply cv_next/server/api/cvs.ts:23]: ok thanks Yechiam

File: cv_next/app/feed/components/feed.tsx (hunk 1 above)
Comment (nevogithub) [reply cv_next/app/feed/components/feed.tsx:20]: ok

**Comment by Aradon267** [comment]
As discussed in the last weekly, I'll handle things here

**Comment by Aradon267** [comment]
Moved to #167

**Pull request closed without merging**